File: CPP/No645.cc
/**
 * Created by Xiaozhong on 2020/11/2.
 * Copyright (c) 2020/11/2 Xiaozhong. All rights reserved.
 */
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<int> findErrorNums(vector<int> &nums) {
        sort(nums.begin(), nums.end());
        int duplicate = -1, missing = 1;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] == nums[i - 1]) duplicate = nums[i];
            else if (nums[i] > nums[i - 1] + 1) missing = nums[i - 1] + 1;
        }
        return {duplicate, nums.back() != nums.size() ? (int) nums.size() : missing};
    }

    void operator<<(vector<int> &nums) {
        vector<int> ans = findErrorNums(nums);
        cout << ans.front() << " " << ans.back() << endl;
    }
};


int main() {
    Solution solution;
    vector<int> nums = {1, 2, 2, 4};
    solution << nums;
    nums = {1, 1};
    solution << nums;
    nums = {2, 2};
    solution << nums;
    nums = {3, 2, 3, 4, 6, 5};
    solution << nums;
}

File: Java/src/No645.java
/*
 * Copyright (c) 2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

import java.util.Arrays;

public class No645 {
    public int[] findErrorNums(int[] nums) {
        Arrays.sort(nums);
        int sum = nums[0];
        int[] ret = new int[2];
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i] == nums[i - 1]) ret[0] = nums[i];
            sum += nums[i];
        }
        ret[1] = ret[0] - (sum - (1 + nums.length) * nums.length / 2);
        return ret;
    }

    public static void main(String[] args) {
        No645 no645 = new No645();
        int[] nums = {3, 2, 3, 4, 6, 5};
        System.out.println(Arrays.toString(no645.findErrorNums(nums)));
    }
}
